src/liboslexec/llvm_util.cpp

```diff
@@ -1149,62 +1154,40 @@ LLVM_Util::op_return (llvm::Value *retval)
 void
 LLVM_Util::op_memset (llvm::Value *ptr, int val, int len, int align)
 {
-    op_memset(ptr, val, constant(len), align);
+    builder().CreateMemSet (ptr, builder().getInt8((unsigned char)val),
+                            uint64_t(len), (unsigned)align);
 }
 
 
 
 void
 LLVM_Util::op_memset (llvm::Value *ptr, int val, llvm::Value *len, int align)
 {
-    // memset with i32 len
-    // and with an i8 pointer (dst) for LLVM-2.8
-    llvm::Type* types[] = {
-        (llvm::Type *) llvm::PointerType::get(llvm::Type::getInt8Ty(context()), 0),
-        (llvm::Type *) llvm::Type::getInt32Ty(context())
-    };
-
-    llvm::Function* func = llvm::Intrinsic::getDeclaration (module(),
-        llvm::Intrinsic::memset,
-        llvm::ArrayRef<llvm::Type *>(types, sizeof(types)/sizeof(llvm::Type*)));
-
-    // NOTE(boulos): constant(0) would return an i32
-    // version of 0, but we need the i8 version. If we make an
-    // ::constant(char val) though then we'll get ambiguity
-    // everywhere.
-    llvm::Value* fill_val = llvm::ConstantInt::get (context(),
-                                                    llvm::APInt(8, val));
-    // Non-volatile (allow optimizer to move it around as it wishes
-    // and even remove it if it can prove it's useless)
-    llvm::Value *args[5] = {
-        ptr, fill_val, len, constant(align), constant_bool(false)
-    };
-    builder().CreateCall (func, llvm::ArrayRef<llvm::Value*>(args, 5));
+    builder().CreateMemSet (ptr, builder().getInt8((unsigned char)val),
+                            uint64_t(len), (unsigned)align);
```

Comment:
Are you sure this is right? Casting `llvm::Value*` to `uint64_t` isn't what you want, right ?

Reply:
I would also add that it seems confusing to have both this version and the overload above.

Reply:
Quite right. For that one, I just wanted to pass the Value* directly.

Reply:
updated